tests/xmap_test.py

```diff
@@ -335,6 +532,47 @@ def f(x, y):
 
     self.assertAllClose(z, jnp.einsum('nij,njk->nik', x, y))
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": f"_{next(test_counter)}",
+       "lhs_shape": lhs_shape, "rhs_shape": rhs_shape, "spec": spec,
+       "axis_resources": axis_resources, "mesh_data": mesh_data}
+      for test_counter in [it.count()]
```

Comment:
lol, another ninja-level feature. Let-bindings in comprehensions!

Reply:
I can't help myself. I love Python golf.